From 90669de789478fb95ec30b93456aaa82daf6ea38 Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Thu, 25 Jul 2019 14:22:21 -0700
Subject: [PATCH] TRITON-1830 cns's boot/setup.sh fails on useradd if re-run
 Reviewed by: John Levon <john.levon@joyent.com> Approved by: Todd Whiteman
 <todd.whiteman@joyent.com>

---
 boot/setup.sh | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/boot/setup.sh b/boot/setup.sh
index 081fe9d..9a4ec15 100755
--- a/boot/setup.sh
+++ b/boot/setup.sh
@@ -7,7 +7,7 @@
 #
 
 #
-# Copyright (c) 2018, Joyent, Inc.
+# Copyright 2019 Joyent, Inc.
 #
 
 export PS4='[\D{%FT%TZ}] ${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
@@ -46,9 +46,11 @@ CONFIG_AGENT_LOCAL_MANIFESTS_DIRS=$SVC_ROOT
 source /opt/smartdc/boot/lib/util.sh
 sdc_common_setup
 
-echo "Adding CNS user"
-useradd -d /opt/triton/cns -P 'Metadata Reader' cns || \
-    fatal "failed to create user"
+if ! grep '^cns:' /etc/passwd >/dev/null; then
+    echo "Adding CNS user"
+    useradd -d /opt/triton/cns -P 'Metadata Reader' cns || \
+        fatal "failed to create user"
+fi
 
 echo "Installing cns redis"
 svccfg import $SVC_ROOT/smf/manifests/cns-redis.xml
-- 
2.21.0

